acceptance/tests/agent/agent_parses_json_catalog.rb

```diff
@@ -0,0 +1,23 @@
+test_name "C99978: Agent parses a JSON catalog"
+tag 'risk:medium'
+
+require 'json'
+
+step "Refresh the catalog" do
+  on(agents, puppet("agent --test --server #{master}"))
+end
+
+step "Agent parses a JSON catalog" do
+  agents.each do |agent|
+    # Is it there and can we read it?
+    json_catalog = File.join(agent.puppet['client_datadir'], 'catalog',
+                             "#{agent.puppet['certname']}.json")
+    on(agent, "[[ -r #{json_catalog} ]]", {:acceptable_exit_codes => [0]})
+
+    # The catalog file should be parseable JSON
+    on(agent, "cat #{json_catalog} | ruby -rjson -e 'JSON.parse(STDIN.read)'")
```

Comment:
Line 15 just says we have read permission on the file. Line 18 says we can parse it as JSON. Didn't want to try to read something before verifying it is present and readable.